service-commands/scripts/allClockValues.js

```diff
@@ -0,0 +1,230 @@
+      : [],
+  }))
+}
+
+/**
+ * @param {string} discoveryProvider - Discovery Provider endpoint
+ * @param {number} batchSize - Batch Size to use for each request
+ * @returns {Object} trackCids - A object mapping user id to a list of track cids
+ */
+async function getTrackCids(discoveryProvider, batchSize) {
+  const trackCids = {}
+
+  let moreTracksRemaining = true
+  for (let offset = 0; moreTracksRemaining; offset += batchSize) {
+    console.time(`Fetching tracks (${offset} - ${offset + batchSize})`)
+
+    const tracksBatch = (
+      await makeRequest({
+        method: "get",
+        url: "/tracks",
+        baseURL: discoveryProvider,
+        params: { offset, limit: batchSize },
+      })
+    ).data.data
+    moreTracksRemaining = tracksBatch.length === batchSize
+
+    tracksBatch.forEach(({ metadata_multihash, owner_id }) => {
+      trackCids[owner_id] = trackCids[owner_id] || []
+      trackCids[owner_id].push(metadata_multihash)
+    })
+
+    console.timeEnd(`Fetching tracks (${offset} - ${offset + batchSize})`)
+  }
+
+  return trackCids
+}
+
+/**
+ * @param {string} creatorNode - Creator Node endpoint
+ * @param {Array<string>} walletPublicKeys
+ * @returns {Array<Object>} clockValues
+ */
+async function getClockValues(creatorNode, walletPublicKeys) {
+  try {
+    return (
+      await makeRequest({
+        method: "post",
+        url: "/users/batch_clock_status",
+        baseURL: creatorNode,
+        data: { walletPublicKeys },
+      })
+    ).data.users
+  } catch (e) {
+    console.log(`Got ${e} when fetching clock values from ${creatorNode}`)
+    return walletPublicKeys.map((walletPublicKey) => ({
+      walletPublicKey,
+      clock: 0,
+    }))
+  }
+}
+
+/**
+ * @param {string} creatorNode - Creator Node endpoint
+ * @param {Array<string>} cids
+ * @returns {Array<Object>} cidsExist
+ */
+async function getCidsExist(creatorNode, cids) {
+  try {
+    return (
+      await makeRequest({
+        method: "post",
+        url: "/batch_cids_exist",
+        baseURL: creatorNode,
+        data: { cids },
+      })
+    ).data.cids
+  } catch (e) {
+    console.log(`Got ${e} when checking if cids exist in ${creatorNode}`)
+    return cids.map((cid) => ({ cid, exists: false }))
+  }
+}
+
+async function run() {
+  const discoveryProvider = "https://discoveryprovider.staging.audius.co/"
+  // const discoveryProvider = "http://localhost:5000"
+  const trackBatchSize = 500
+  const userBatchSize = 500
+
+  const trackCids = await getTrackCids(discoveryProvider, trackBatchSize)
+
+  let moreUsersRemaining = true
+  for (let offset = 0; moreUsersRemaining; offset += userBatchSize) {
+    console.time(`${offset} - ${offset + userBatchSize}`)
+
+    const usersBatch = await getUsersBatch(
+      discoveryProvider,
+      offset,
+      userBatchSize
+    )
+    moreUsersRemaining = usersBatch.length === userBatchSize
+
+    const creatorNodes = new Set()
+    const cn2wallets = {} // creator node -> wallets
```

Comment:
why is this cn2-specific? or just a relic from the staging run?

Reply:
Sorry, this was actually meant to refer to `creator node to wallets`, I will rename this to something more obvious.